CITATION.cff

```diff
@@ -0,0 +1,75 @@
+cff-version: 1.2.0
+message: "If you use this software, please cite it as below."
+authors:
+- family-names: "Krech"
+  given-names: "Daniel"
+- family-names: "Grimnes"
+  given-names: "Gunnar AAstrand"
+- family-names: "Higgins"
+  given-names: "Graham"
+- family-names: "Hees"
+  given-names: "Jörn"
+  orcid: "https://orcid.org/0000-0002-0084-8998"
+- family-names: "Aucamp"
+  given-names: "Iwan"
+  orcid: "https://orcid.org/0000-0002-7325-3231"
+- family-names: "Lindström"
+  given-names: "Niklas"
+- family-names: "Arndt"
+  given-names: "Natanael"
+  orcid: "https://orcid.org/0000-0002-8130-8677"
+- family-names: "Sommer"
+  given-names: "Ashley"
+  orcid: "https://orcid.org/0000-0003-0590-0131"
+- family-names: "Chuc"
+  given-names: "Edmond"
+  orcid: "https://orcid.org/0000-0002-6047-9864"
+- family-names: "Herman"
+  given-names: "Ivan"
+  orcid: "https://orcid.org/0000-0003-0782-2704"
+- family-names: "Nelson"
+  given-names: "Alex"
+- family-names: "McCusker"
+  given-names: "Jamie"
+  orcid: "https://orcid.org/0000-0003-1085-6059"
+- family-names: "Gillespie"
+  given-names: "Tom"
+- family-names: "Kluyver"
+  given-names: "Thomas"
+  orcid: "https://orcid.org/0000-0003-4020-6364"
+- family-names: "Ludwig"
+  given-names: "Florian"
+- family-names: "Champin"
+  given-names: "Pierre-Antoine"
+  orcid: "https://orcid.org/0000-0001-7046-4474"
+- family-names: "Watts"
+  given-names: "Mark"
+- family-names: "Holzer"
+  given-names: "Urs"
+- family-names: "Summers"
+  given-names: "Ed"
+- family-names: "Morriss"
+  given-names: "Whit"
+- family-names: "Winston"
+  given-names: "Donny"
+- family-names: "Perttula"
+  given-names: "Drew"
+- family-names: "Kovacevic"
+  given-names: "Filip"
+  orcid: "https://orcid.org/0000-0002-2854-0434"
+- family-names: "Chateauneu"
+  given-names: "Remi"
+  orcid: "https://orcid.org/0000-0002-7505-8149"
+- family-names: "Solbrig"
+  given-names: "Harold"
+  orcid: "https://orcid.org/0000-0002-5928-3071"
+- family-names: "Cogrel"
+  given-names: "Benjamin"
+  orcid: "https://orcid.org/0000-0002-7566-4077"
+- family-names: "Stuart"
+  given-names: "Veyndan"
+title: "RDFLib"
+version: 6.3.2
+date-released: 2023-03-26
+url: "https://github.com/RDFLib/rdflib"
+doi: 10.5281/zenodo.7771749
```

Comment:
```suggestion
doi: 10.5281/zenodo.6845245
```

The DOI is a bit of a problem.

![image](https://github.com/RDFLib/rdflib/assets/4481387/4bf34cf6-5388-416f-88ec-88e20bb01511)

The above is from https://zenodo.org/account/settings/github/, how I interpret it, and how it mostly has worked, is that after creating a release using GitHub releases, Zenodo fetches the archive of the release (which for 6.3.2 is https://github.com/RDFLib/rdflib/tree/0593c75753b0b80256d448d56328dbc3967cb055) and then assigns a DOI and publishes the archive under the DOI.

The `CITATION.cff` file should be taken from the release archive, but before Zenodo publishes it we don't know the version-specific DOI.

However, we do have a DOI which is not version-specific which Zenodo refers to as the [concept DOI](https://help.zenodo.org/faq/#hversioning-3). For RDFLib it is https://doi.org/10.5281/zenodo.6845245. I think if we put any value in here, the best is to put https://doi.org/10.5281/zenodo.6845245 in here. The CITATION.cff documentation have an example of both using the versioned and concept DOI in here [[ref](https://github.com/citation-file-format/citation-file-format/blob/main/schema-guide.md#doi)], so I think it is acceptable.